.github/workflows/push-ghcr.yaml

```diff
@@ -0,0 +1,52 @@
+name: Create and publish a Docker image
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches: ['main']
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push Primaza Docker image
+        run: |
+            IMG=$IMG_BASE:$(git rev-parse --short HEAD) make primaza docker-build docker-push
+            docker tag "$IMG_BASE:$(git rev-parse --short HEAD)" "$IMG_BASE:latest"
+            docker push "$IMG_BASE:latest"
+        env:
+          IMG_BASE: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+
+      - name: Build and push AgentApp Docker image
+        run: |
+            IMG=$IMG_BASE:$(git rev-parse --short HEAD) make agentapp docker-build docker-push
+            docker tag "$IMG_BASE:$(git rev-parse --short HEAD)" "$IMG_BASE:latest"
+            docker push "$IMG_BASE:latest"
+        env:
+          IMG_BASE: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}-agentapp
+
+      - name: Build and push AgentSvc Docker image
+        run: |
+            IMG=$IMG_BASE:$(git rev-parse --short HEAD) make agentsvc docker-build docker-push
+            docker tag "$IMG_BASE:$(git rev-parse --short HEAD)" "$IMG_BASE:latest"
+            docker push "$IMG_BASE:latest"
+        env:
+          IMG_BASE: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}-agentsvc
```

Comment:
We should push all images at once: if an image fails to build, none of the other images for that commit to be pushed into the image repository.